### ISSUE TITLE
docs: auto-generate configuration options for each lint rule

### ISSUE THREAD

**Issue by camchenry** [oxc-project/oxc#9617]
currently, the website is missing almost all of the configuration options for lint rules generally, unless we remembered to include this in the description. 

Since most lint rules have a configuration struct specific to the rule, and a struct field for each option, maybe we could auto-generate these?

**Comment by therewillbecode** [comment]
Do you mean we should be auto generating the doc comments for the option configuration by parsing the configuration struct for each rule?

**Comment by camchenry** [comment]
@therewillbecode I'm not completely sure on this one yet, I don't think it's clear to me. I think something that like allowing annotations on a config struct, or perhaps a trait that can be implemented like:

```rs
pub struct RuleConfig {
  /// Allows accepting some cases
  ///
  /// Example: ...
  #[rename = "someOption"]
  #[type = bool, default = false]
  some_option: serde_json::Value,
}
impl Rule for RuleName {
  type Config = RuleConfig

  // ...
}
```

and then using the `Config` type and generating the docs from that like:

```md
## Option: `someOption` (default: false)

Allows accepting some cases.

Example: ...
```

Maybe there is a good way to use macros for this, but I'm not much of a macro expert. Or else we could use serde directly maybe? If anyone else has ideas on this, I'd be interested.
